tests/actions/PolicyTest.js

```diff
@@ -0,0 +1,197 @@
+                .then(
+                    () =>
+                        new Promise((resolve) => {
+                            const connectionID = Onyx.connect({
+                                key: ONYXKEYS.COLLECTION.REPORT,
+                                waitForCollectionCallback: true,
+                                callback: (allReports) => {
+                                    Onyx.disconnect(connectionID);
+
+                                    // Three reports should be created: #announce, #admins and expense report
+                                    const workspaceReports = _.filter(allReports, (report) => report.policyID === policyID);
+                                    expect(_.size(workspaceReports)).toBe(3);
+                                    _.forEach(workspaceReports, (report) => {
+                                        expect(report.pendingFields.addWorkspaceRoom).toBe(CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
+                                        expect(report.participantAccountIDs).toEqual([ESH_ACCOUNT_ID]);
+                                        switch (report.chatType) {
+                                            case CONST.REPORT.CHAT_TYPE.POLICY_ADMINS: {
+                                                adminReportID = report.reportID;
+                                                break;
+                                            }
+                                            case CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE: {
+                                                announceReportID = report.reportID;
+                                                break;
+                                            }
+                                            case CONST.REPORT.CHAT_TYPE.POLICY_EXPENSE_CHAT: {
+                                                expenseReportID = report.reportID;
+                                                break;
+                                            }
+                                            default: break;
+                                        }
+                                    });
+
+                                    resolve();
+                                },
+                            });
+                        }),
+                )
+                .then(
+                    () =>
+                        new Promise((resolve) => {
+                            const connectionID = Onyx.connect({
+                                key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+                                waitForCollectionCallback: true,
+                                callback: (reportActions) => {
+                                    Onyx.disconnect(connectionID);
+
+                                    // Three report actions should have CREATED action: #announce, #admins and expense report
+                                    const adminReportActions = _.filter(reportActions, (reportAction) => reportAction.reportID === adminReportID);
```

Comment:
I think you can maybe just do this?

```diff
- const adminReportActions = _.filter(reportActions, (reportAction) => reportAction.reportID === adminReportID);
+ const adminReportActions = reportActions[adminReportID];
```

Bit simpler 🙂

Reply:
<img width="657" alt="Screenshot 2024-01-08 at 6 12 21 PM" src="https://github.com/Expensify/App/assets/77237602/8b80ef11-5aba-4d97-b884-d6b66992fdce">
This is how the action looks like 
So we would have to do something like this:
```js
const adminReportActions = _.values(reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${adminReportID}`]);
```